File: leetcode_c_plus_plus/504.cpp
// 504. 七进制数
// https://leetcode-cn.com/problems/base-7/submissions/
#include<map>
#include<set>
#include<vector>
#include<cmath>
#include<numeric>
#include<unordered_map>
using namespace std;
#define INT_MAX 2147483647
#define INT_MIN (-INT_MAX - 1)
/**
@File : 504.cpp
@Time : 2022/03/07 23:04:28
@Author : YuMin Zhang
@State : Indepeedent | Half-Depedent | Depedent 
@Thinking :
@Tag : Easy | Medium | Hard
**/
class Solution {
public:
    string convertToBase7(int num) {
        if(num == 0) return "0";
        bool flag = false;
        if(num < 0) {
            flag = true;
            num = abs(num);
        }
        string ans = "";
        while(num > 0) {
            ans += to_string(num % 7);
            num = num / 7;
        }
        if(flag) ans += "-";
        reverse(ans.begin(), ans.end());
        return ans;

    }
};

File: leetcode_python/504.py
# 504. 七进制数
# https://leetcode-cn.com/problems/base-7/
from typing import List
from collections import defaultdict
from collections import deque
from math import inf
import bisect
import heapq
'''
@File : 504.py
@Time : 2022/03/07 22:59:40
@Author : YuMin Zhang
@State : Indepeedent | Half-Depedent | Depedent 
@Thinking :
@Tag : Easy | Medium | Hard
'''
class Solution:
    def convertToBase7(self, num: int) -> str:
        ans = []
        flag = False
        if num < 0:
            flag = True
            num = abs(num)
        if num == 0:
            return "0"    
        while num > 0:
            bit = num % 7
            num = num // 7
            ans.append(str(bit))
        if flag:
            ans.append("-")
        ans.reverse()
        return "".join(ans)